crates/bevy_ui/src/widget/button.rs

```diff
@@ -1,7 +1,94 @@
-use bevy_ecs::{prelude::Component, reflect::ReflectComponent};
+use bevy_ecs::prelude::*;
+use bevy_hierarchy::{Children, HierarchyQueryExt};
+use bevy_picking::{
+    events::{Down, Pointer, Up},
+    focus::HoverMap,
+    pointer::PointerId,
+};
 use bevy_reflect::{std_traits::ReflectDefault, Reflect};
 
-/// Marker struct for buttons
+/// The data required to create a buttonlike widget that can respond to hovering and being clicked / pressed.
+///
+/// Information about whether or not this button is currently pressed or hovered is gathered from
+/// user input, processed via [`bevy_picking`] and set during [`UiSystem::Focus`](crate::UiSystem::Focus).
+///
+/// Use the [`Pickable`](bevy_picking::Pickable) component to configure whether or not the button should block interactions with lower nodes.
 #[derive(Component, Debug, Default, Clone, Copy, PartialEq, Eq, Reflect)]
 #[reflect(Component, Default, Debug, PartialEq)]
-pub struct Button;
+pub struct Button {
+    /// Whether the button is currently pressed.
+    ///
+    /// This is `true` as long as the button is pressed, but only while the pointer
+    /// is over the button.
+    ///
+    /// This value is set by the [`button_down_observer`] and [`button_up_observer`] [`Observer`]s.
+    pub pressed: bool,
+    /// Whether the button is currently hovered.
+    ///
+    /// This is only `true` while the pointer is over this button,
+    /// and will not be set if there is no pointer.
+    ///
+    /// This value is set by the [`update_hover_status`] system.
+    pub hovered: bool,
+}
+
+/// Updates the [`Button::hovered`] field based on the current [`HoverMap`] data.
+///
+/// Only [`PointerId::Mouse`] data is considered.
+/// A button will be considered hovered if the mouse is over the button itself or any of its descendants.
+pub fn update_hover_status(
+    mut button_query: Query<(Entity, &mut Button)>,
+    hover_map: Res<HoverMap>,
+    children_query: Query<&Children>,
+) {
+    let Some(map) = hover_map.get(&PointerId::Mouse) else {
+        // If no appropriate hover map data is found, we can assume that no buttons are hovered
+        for (_, mut button) in button_query.iter_mut() {
+            // Avoid triggering change detection spuriously
+            // We can't use `set_if_neq` here because we're only looking at a single component
+            if button.hovered {
+                button.hovered = false;
+            }
```

Comment:
```suggestion
            button.set_if_neq(Button {
                pressed: button.pressed,
                hovered: false,
            });
```

Reply:
Wait, if no buttons are hovered  we don't need to preserve the `pressed` state either, so both fields should be set to false?